Core/clim-basic/ports.lisp

```diff
@@ -385,13 +406,15 @@
           (set-sheet-pointer-cursor port event-sheet new-pointer-cursor))))
     ;; Handle some events specially.
     (typecase event
-      ;; Pressing pointer button over a sheet makes a sheet
-      ;; pressed. Pressed sheet is assigned only when there is
-      ;; currently none.
+      ;; Pressing pointer button over a sheet makes a sheet pressed
+      ;; and focused. Pressed sheet is assigned only when there is
+      ;; currently none while focused event is always sent.
       (pointer-button-press-event
        (when (null pressed-sheet)
-         (setf (port-pressed-sheet port) new-pointer-sheet)))
-      ;; Releasing the button sets the pressed sheet to NIL.
+         (setf (port-pressed-sheet port) new-pointer-sheet))
+       (dispatch-event-copy new-pointer-sheet event 'window-manager-focus-event))
```

Comment:
I'm not sure about the "window manager" part in `window-manager-focus-event`. Isn't `window-manager-focus-event` for the case in which a frame gets the focus on the window manager/desktop environment level?

Reply:
in principle the top-level-sheet receives this event when it is focused (on CLX; this event is not specified in CLIM II), In principle we could have enabled other mirrored sheets to receive such event (but we don't do that).It is used only to implement input focus, so I see no reason to inflate number of different events (like we do not introduce separate class for synthesized boundary events).